src/modal-context/modal-context.tsx

```diff
@@ -0,0 +1,40 @@
+import { useShowModal } from 'hooks';
+import React, { createContext, useState } from 'react';
+
+export const ModalsContext = createContext(undefined);
+
+const ModalsContextProvider = ({ children }) => {
+  const [movieDetails, setMovieDetails] = useState(null);
+  const [loadingMovieDetails, setLoadingMovieDetails] = useState(true);
+  const [errorMovieDetails, setErrorMovieDetails] = useState(false);
```

Comment:
When you update one of these states, all the children will be rendered. You are wrapping the whole application with this provider so everything will be rerendered, which is not good because you only need to open the modal.
The solution could be to use redux instead of plane context